### PR TITLE
Add Liquidity Migrator to help migrate LP tokens to Kyber DMM

### PR DESCRIPTION
**Pull request by manhlx3006**

### CODE REVIEW

**Review by Anyhowclick — APPROVED** [review]

File: contracts/wrapper/LiquidityMigrator.sol [hunk 1]
```diff
@@ -0,0 +1,400 @@
...
+        require(amountB >= amountBMin, "Migratior: UNI_INSUFFICIENT_B_AMOUNT");
+    }
+
+    /**
+    * @dev only approve if the current allowance is 0
+    */
+    function _safeApproveAllowance(IERC20 token, address spender, uint256 allowance)
+        internal
+    {
+        if (token.allowance(address(this), spender) == 0) {
```
Comment (ducquangkstn) [inline contracts/wrapper/LiquidityMigrator.sol:462]: There is an edge case when Migrator approve to amountADesired and only transfer amountAMin. So the allowance would be amountADesired - amountAMin. 
The safeway to do it here is approving to max (-1)

File: contracts/wrapper/LiquidityMigrator.sol [hunk 2]
```diff
@@ -0,0 +1,400 @@
...
+    using SafeMath for uint256;
+
+    address immutable public dmmRouter;
+
+    constructor(address _dmmRouter) public {
+        require(_dmmRouter != address(0), "Migrator: INVALID_ROUTER");
+        dmmRouter = _dmmRouter;
+    }
+
+    receive() external payable {}
```
Comment (ducquangkstn) [inline contracts/wrapper/LiquidityMigrator.sol]: we don't need this, right?

File: contracts/wrapper/LiquidityMigrator.sol [hunk 3]
```diff
@@ -0,0 +1,400 @@
...
+        uint256 addedLiquidity
+    ) {
+        // support for both normal token and token with fee on transfer
+        {
+            uint256 balanceTokenA = IERC20(tokenA).balanceOf(address(this));
+            uint256 balanceTokenB = IERC20(tokenB).balanceOf(address(this));
+            (amountA, amountB) = _removeUniLiquidity(
+                uniPair, tokenA, tokenB, liquidity, amountAMin, amountBMin, deadline
+            );
+            amountA = IERC20(tokenA).balanceOf(address(this)).sub(balanceTokenA);
```
Comment (ducquangkstn) [inline contracts/wrapper/LiquidityMigrator.sol]: either we handle the case for fee-on-transfer token or we remove the amountA, amountB returned from _removeLiquidity

File: contracts/wrapper/LiquidityMigrator.sol (hunk 1 above)
Comment (manhlx3006) [reply contracts/wrapper/LiquidityMigrator.sol:462]: already changed to approve for max value

File: contracts/wrapper/LiquidityMigrator.sol (hunk 3 above)
Comment (manhlx3006) [reply contracts/wrapper/LiquidityMigrator.sol]: yeah can remove the return values

File: contracts/wrapper/LiquidityMigrator.sol (hunk 2 above)
Comment (manhlx3006) [reply contracts/wrapper/LiquidityMigrator.sol]: yeah can remove the receive func

**Review by ducquangkstn — APPROVED** [review]

**Pull request merged**